tests/unittests/algo/test_tpe.py

```diff
@@ -325,20 +496,23 @@ def test_sample_real_dimension(self):
         space.register(dim2)
 
         tpe = TPE(space)
-        points = numpy.random.uniform(-10, 10, 20).reshape(20, 1)
-        below_points = points[:6, :]
-        above_points = points[6:, :]
-        points = tpe.sample_real_dimension(dim1, 1, below_points, above_points)
+        points = numpy.random.uniform(-10, 10, 20)
+        below_points = [points[:8]]
+        above_points = [points[8:]]
+        points = tpe.sample_one_dimension(dim1, 1,
+                                          below_points, above_points, tpe._sample_real_point)
         assert len(points) == 1
 
-        points = numpy.random.uniform(-5, 5, 32).reshape(16, 2)
-        below_points = points[:4, :]
-        above_points = points[4:, :]
-        points = tpe.sample_real_dimension(dim2, 2, below_points, above_points)
+        points = numpy.random.uniform(-5, 5, 32)
+        below_points = [points[:8], points[8:16]]
+        above_points = [points[16:24], points[24:]]
+        points = tpe.sample_one_dimension(dim2, 2,
+                                          below_points, above_points, tpe._sample_real_point)
         assert len(points) == 2
 
         tpe.n_ei_candidates = 0
-        points = tpe.sample_real_dimension(dim2, 2, below_points, above_points)
+        points = tpe.sample_one_dimension(dim2, 2,
+                                          below_points, above_points, tpe._sample_real_point)
```

Comment:
I see that we didn't test for values for test_sample_real_dimensions. We could sample separately below and above with different intervals to artificially create the best regions and verify the sampling of the TPE.

Reply:
It was same idea for int, categorical and real, as there are separate cases for gmm and categorical sampler. But sure, we can add some specific verification at these places.